src/main/java/gregtech/api/cover/CoverSaveHandler.java

```diff
@@ -0,0 +1,149 @@
+package gregtech.api.cover;
+
+import gregtech.api.util.GTLog;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.BiConsumer;
+
+public final class CoverSaveHandler {
+
+    public static final int NO_COVER_ID = -1;
+
+    private CoverSaveHandler() {}
+
+    /**
+     * Write sync data for a CoverableView's covers
+     *
+     * @param buf           the buf to write to
+     * @param coverableView the CoverableView containing the covers
+     */
+    public static void writeInitialSyncData(@NotNull PacketBuffer buf, @NotNull CoverableView coverableView) {
+        for (EnumFacing coverSide : EnumFacing.VALUES) {
+            Cover cover = coverableView.getCoverAtSide(coverSide);
+            if (cover != null) {
+                buf.writeVarInt(CoverDefinition.getNetworkIdForCover(cover.getDefinition()));
+                cover.writeInitialSyncData(buf);
+            } else {
+                // cover was not attached
+                buf.writeVarInt(NO_COVER_ID);
```

Comment:
Do we need to write if no cover was attached to the CoverableView? To me that just seems like a bit of network spam. We could also change the structure of the sync data a little bit to be a bit smarter in the receive section.

If we write the data as: EnumFacing.ordinal, cover id, cover sync data, we could avoid looping over all the EnumFacings when receiving the data packet, and instead just specifically interact with the facings that have a cover (would have to just retreive the facings by ordinal from the enum). We already do something similar in the coverPlacement networking.